DataOnion/Extensions.cs

```diff
@@ -1,47 +1,95 @@
+            Func<string, T> connectionGetter
+        ) where T : DbConnection;
+    }
+
+    public class FluentDatabaseOnion : IFluentDatabaseOnion, IFluentDapperSetup, IFluentEfCoreSetup
+    {
+        private readonly IServiceCollection _serviceCollection;
+        private readonly string _connectionString;
+
+        public FluentDatabaseOnion(
+            IServiceCollection serviceCollection,
+            string connectionString
         )
-            where TDbContext : DbContext
         {
-            serviceCollection.AddDbContext<TDbContext>(
-                options?.DataConnector?.Invoke(options.ConnectionString),
-                options?.ServiceLifetime ?? ServiceLifetime.Scoped,
-                options?.OptionsLifetime ?? ServiceLifetime.Scoped
+            _serviceCollection = serviceCollection;
+            _connectionString = connectionString;
+        }
+
+        public IFluentDapperSetup ConfigureDapper<T>(
+            Func<string, T> connectionGetter
+        ) where T: DbConnection
+        {
+            _serviceCollection.AddScoped<T>(
+                services => connectionGetter(_connectionString)
+            );
+
+            return this;
+        }
+
+        public IFluentEfCoreSetup ConfigureEfCore<T>(
+            Func<string, Action<DbContextOptionsBuilder>>? dataConnector,
+            ServiceLifetime contextLifetime = ServiceLifetime.Scoped,
+            ServiceLifetime optionsLifetime = ServiceLifetime.Scoped
+        ) where T : DbContext
+        {
+            _serviceCollection.AddDbContext<T>(
+                dataConnector?.Invoke(_connectionString),
+                contextLifetime,
+                optionsLifetime
             );
 
-            serviceCollection.AddScoped<IEFCoreService<TDbContext>, EFCoreService<TDbContext>>();
+            _serviceCollection.AddScoped<IEFCoreService<T>, EFCoreService<T>>();
 
-            return serviceCollection;
+            return this;
         }
+    }
 
-        public static IServiceCollection ConfigureDapperOnion<TDbConnection>(
+    public static class DataHelperExtensions
+    {
+        public static IFluentDatabaseOnion? DatabaseOnion;
+        public static IFluentDatabaseOnion AddDatabaseOnion(
             this IServiceCollection serviceCollection,
-            DapperOptions<TDbConnection>? options = null
+            string connectionString
         )
-            where TDbConnection : DbConnection
         {
-            // If no options are passed in, assume the DBConnection has already been registered
-            if (options != null)
-            {
-                serviceCollection.AddScoped<TDbConnection>(
-                    services => options.ConnectionGetter(options.ConnectionString)
-                );
-            }
-
-            serviceCollection.AddScoped<IDapperService<TDbConnection>, DapperService<TDbConnection>>();
-            
-            return serviceCollection;
+            DatabaseOnion = new FluentDatabaseOnion(
```

Comment:
Why is this being stored in a static property? Is anything lost from just `return new ...`?

Reply:
it allows us to retain the object for internal use by the library later if we needed to